spring-boot/proxy/src/test/java/eu/coatrack/proxy/security/local_api_key_manager_tests/GatewayModeUpdateLoggingTest.java

```diff
@@ -0,0 +1,151 @@
+package eu.coatrack.proxy.security.local_api_key_manager_tests;
```

Comment:
These test classes are cohesive. IMHO removing the sub-folder would look untidy and confusing, since the test classes are scattered between a bunch of other test classes they share no business with.

What would be the advantage of not using the sub-folder?

Reply:
I would consider it best practice that you can find the test classes in the same package as the class being tested. 

In addition, I think it is not ideal that the package name is derived from the name of another class, which means that if you would change the class name in the future, this relationship could become unclear.

It would be better to prefix the names of all test classes with the name of the class being tested, then you can easily see which test classes belong together and how is their relationship to the classes being tested.

Reply:
Good points and your strategy makes sense. Done as you suggested.